Modules/ITS/src/TrendingTaskITSFEE.cxx

```diff
@@ -0,0 +1,356 @@
+  }
+}
+
+// todo: see if OptimizeBaskets() indeed helps after some time
+void TrendingTaskITSFEE::update(Trigger t, framework::ServiceRegistry& services)
+{
+  auto& qcdb = services.get<repository::DatabaseInterface>();
+
+  trendValues(t, qcdb);
+  storePlots(qcdb);
+  storeTrend(qcdb);
+}
+
+void TrendingTaskITSFEE::finalize(Trigger t, framework::ServiceRegistry& services)
+{
+  auto& qcdb = services.get<repository::DatabaseInterface>();
+
+  storePlots(qcdb);
+  storeTrend(qcdb);
+}
+
+void TrendingTaskITSFEE::storeTrend(repository::DatabaseInterface& qcdb)
+{
+  auto mo = std::make_shared<core::MonitorObject>(mTrend.get(), getName(), "o2::quality_control_modules::its::TrendingTaskITSFEE",
+                                                  mConfig.detectorName);
+  mo->setIsOwner(false);
+  qcdb.storeMO(mo);
+}
+
+void TrendingTaskITSFEE::trendValues(const Trigger& t, repository::DatabaseInterface& qcdb)
+{
+  // We use current date and time. This for planned processing (not history). We
+  // still might need to use the objects
+  // timestamps in the end, but this would become ambiguous if there is more
+  // than one data source.
+
+  mTime = TDatime().Convert();
+  mMetaData.runNumber = t.activity.mId;
+  int count = 0;
+
+  for (auto& dataSource : mConfig.dataSources) {
+    // todo: make it agnostic to MOs, QOs or other objects. Let the reductor
+    // cast to whatever it needs.
+    if (dataSource.type == "repository") {
+      auto mo = qcdb.retrieveMO(dataSource.path, dataSource.name, t.timestamp, t.activity);
+      if (!count) {
+        std::map<std::string, std::string> entryMetadata = mo->getMetadataMap();  // full list of metadata as a map
+        mMetaData.runNumber = std::stoi(entryMetadata[metadata_keys::runNumber]); // get and set run number
+        runlist.push_back(std::to_string(mMetaData.runNumber));
+      }
+      TObject* obj = mo ? mo->getObject() : nullptr;
+      if (obj)
+        mReductors[dataSource.name]->update(obj);
+    } else {
+      ILOGE << "Unknown type of data source '" << dataSource.type << "'.";
+    }
+    count++;
+  }
+  mTrend->Fill();
+}
+
+void TrendingTaskITSFEE::storePlots(repository::DatabaseInterface& qcdb)
+{
+  // ILOG(Info, Support) << "Generating and storing " << mConfig.plots.size() << " plots." << ENDM;
+
+  int countplots = 0;
+  int countITSpart = 0;
+  bool isRun = false;
+
+  // Define output canvas and legend
+  TCanvas* canvas;
+  TLegend* legend;
+
+  TMultiGraph* multi_trend;
+  TGraph* trend_plot_Flags[nFlags];
+
+  // Retrieve X-axis for trend plot. Two options: "Time" or "Run"
+  long int numberOfEntries = mTrend->GetEntriesFast();
+  mTrend->Draw("time", "", "goff");
+  std::vector<double> retrieveTime(mTrend->GetV1(), mTrend->GetV1() + mTrend->GetSelectedRows() % mTrend->GetEstimate());
+
+  std::vector<double> retrieveRunNumber;
+  for (int j = 1; j <= numberOfEntries; j++)
+    retrieveRunNumber.push_back((double)j);
```

Comment:
Would it be possible to unify construction of the x-axis? I would use as an example FHR task approach where we obtain X/Y axis with mTrend->GetV1() and mTrend->GetV2() function